src/App.tsx

```diff
@@ -1,61 +1,129 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoList } from './components/TodoList';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import users from './api/users';
+
+const getUserById = (usersArr: User[], userId: number) => {
+  return usersArr.find(user => user.id === userId) || null;
+};
+
+const toDoWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(usersFromServer, todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(toDoWithUser);
+
+  const [title, setTitle] = useState<string>('');
+  const [hasTitleError, setHasTitleError] = useState<boolean>(false);
+
+  const [userId, setUserId] = useState<number>(0);
+  const [hasUserIdError, setHasUserIdError] = useState<boolean>(false);
+
+  const onTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const onUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setHasTitleError(true);
+    }
+
+    if (!userId) {
+      setHasUserIdError(true);
+    }
+
+    if (title && userId) {
+      const maxId =
+        todos.length > 0 ? Math.max(...todos.map(todo => todo.id)) : 0;
+
+      const newTodo: Todo = {
+        id: maxId + 1,
+        title,
+        userId,
+        user: getUserById(users, userId),
```

Comment:
The `getUserById` function is called with `users` instead of `usersFromServer`. Since `users` is not used elsewhere and is redundant, replace `users` with `usersFromServer` to maintain consistency and avoid potential errors.